diff --git a/chrome/android/java/res/xml/privacy_preferences.xml b/chrome/android/java/res/xml/privacy_preferences.xml
--- a/chrome/android/java/res/xml/privacy_preferences.xml
+++ b/chrome/android/java/res/xml/privacy_preferences.xml
@@ -61,5 +61,10 @@
         android:summary="@string/preload_pages_summary"
         android:fragment="org.chromium.chrome.browser.prefetch.settings.PreloadPagesSettingsFragment" />
+    <org.chromium.components.browser_ui.settings.ChromeSwitchPreference
+        android:key="tabgrid_use_icons"
+        android:title="@string/tabgrid_use_icons_title"
+        android:summary="@string/tabgrid_use_icons_summary"
+        android:defaultValue="false" />
     <org.chromium.chrome.browser.incognito.reauth.IncognitoReauthSettingSwitchPreference
         android:key="incognito_lock"
         android:title="@string/settings_incognito_tab_lock_title"
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/privacy/settings/PrivacySettings.java b/chrome/android/java/src/org/chromium/chrome/browser/privacy/settings/PrivacySettings.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/privacy/settings/PrivacySettings.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/privacy/settings/PrivacySettings.java
@@ -9,2 +9,3 @@
 import android.os.Build;
+import android.content.SharedPreferences;
 import android.os.Bundle;
@@ -46,2 +47,8 @@
 import org.chromium.components.browser_ui.settings.SettingsUtils;
+import org.chromium.base.ContextUtils;
+import org.chromium.chrome.browser.lifetime.ApplicationLifetime;
+import org.chromium.chrome.browser.ui.messages.snackbar.INeedSnackbarManager;
+import org.chromium.chrome.browser.ui.messages.snackbar.Snackbar;
+import org.chromium.chrome.browser.ui.messages.snackbar.SnackbarManager;
+import java.util.function.Supplier;
 import org.chromium.components.browser_ui.site_settings.ContentSettingsResources;
@@ -56,6 +63,16 @@
 /** Fragment to keep track of the all the privacy related preferences. */
 public class PrivacySettings extends ChromeBaseSettingsFragment
-        implements Preference.OnPreferenceChangeListener {
+        implements Preference.OnPreferenceChangeListener, INeedSnackbarManager {
+
+    private Supplier<SnackbarManager> mSnackbarManagerSupplier;
+    private Snackbar mSnackbar;
+
+    @Override
+    public void setSnackbarManagerSupplier(Supplier<SnackbarManager> manager) {
+        mSnackbarManagerSupplier = manager;
+    }
+
+    public static final String PREF_TABGRID_USE_ICONS = "tabgrid_use_icons";
     private static final String PREF_CAN_MAKE_PAYMENT = "can_make_payment";
     private static final String PREF_PRELOAD_PAGES = "preload_pages";
     private static final String PREF_HTTPS_FIRST_MODE = "https_first_mode";
@@ -124,6 +133,24 @@
         }
 
         Preference privacyGuidePreference = findPreference(PREF_PRIVACY_GUIDE);
+        mSnackbar = Snackbar.make(getActivity().getString(R.string.ui_relaunch_notice),
+                new SnackbarManager.SnackbarController() {
+                        @Override
+                        public void onDismissNoAction(Object actionData) { }
+
+                        @Override
+                        public void onAction(Object actionData) {
+                                ApplicationLifetime.terminate(true);
+                        }
+                }, Snackbar.TYPE_NOTIFICATION, Snackbar.UMA_UNKNOWN)
+                .setDefaultLines(false)
+                .setAction(getActivity().getString(R.string.relaunch),
+                        /*actionData*/null)
+                .setDuration(/*durationMs*/70000);
+
+        Preference tabGridUseIcons = findPreference(PREF_TABGRID_USE_ICONS);
+        tabGridUseIcons.setOnPreferenceChangeListener(this);
+
         // Record the launch of PG from the S&P link-row entry point
         privacyGuidePreference.setOnPreferenceClickListener(
                 preference -> {
@@ -290,6 +316,10 @@
         } else if (PREF_HTTPS_FIRST_MODE.equals(key)) {
             UserPrefs.get(getProfile())
                     .setBoolean(Pref.HTTPS_ONLY_MODE_ENABLED, (boolean) newValue);
+        } else if (PREF_TABGRID_USE_ICONS.equals(key)) {
+            SharedPreferences.Editor sharedPreferencesEditor = ContextUtils.getAppSharedPreferences().edit();
+            sharedPreferencesEditor.putBoolean(PREF_TABGRID_USE_ICONS, (boolean)newValue);
+            sharedPreferencesEditor.apply();
         }
         return true;
     }
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java b/chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java
@@ -138,4 +138,5 @@
 import org.chromium.chrome.browser.preferences.ChromeSharedPreferences;
 import org.chromium.chrome.browser.preferences.Pref;
+import org.chromium.chrome.browser.privacy.settings.PrivacySettings;
 import org.chromium.chrome.browser.printing.TabPrinter;
 import org.chromium.chrome.browser.profiles.Profile;
@@ -706,7 +706,7 @@
                    new TabContentManager(
                            this,
                            mBrowserControlsManagerSupplier.get(),
-                            !SysUtils.isLowEndDevice(),
+                            !SysUtils.isLowEndDevice() && !ContextUtils.getAppSharedPreferences().getBoolean(PrivacySettings.PREF_TABGRID_USE_ICONS, false),
                            tabModelSelector != null ? tabModelSelector::getTabById : null,
                            TabWindowManagerSingleton.getInstance()));
 
diff --git a/chrome/browser/ui/android/strings/android_chrome_strings.grd b/chrome/browser/ui/android/strings/android_chrome_strings.grd
--- a/chrome/browser/ui/android/strings/android_chrome_strings.grd
+++ b/chrome/browser/ui/android/strings/android_chrome_strings.grd
@@ -5741,5 +5741,13 @@
 
+      <!-- Tab Broup Disable Thumbnail -->
+      <message name="IDS_TABGRID_USE_ICONS_TITLE" desc="Text for 'Use icons in the tab grid' settings-privacy option.">
+        Use only icons in the tab grid
+      </message>
+      <message name="IDS_TABGRID_USE_ICONS_SUMMARY" desc="Summary text for 'Use icons in the tab grid' settings-privacy option.">
+        Disable saving snapshots to disk (even for incognito sessions)
+      </message>
+
       <!-- Tab Group Tab Switcher strings -->
       <message name="IDS_TAB_MANAGEMENT_MODULE_TITLE" desc="Text shown when the Tab Management module is referenced in install start, success, failure UI (e.g. in IDS_MODULE_INSTALL_START_TEXT, which will expand to 'Installing Tab management for Chrome...').">
         Tab management
       </message>
--
